dll/win32/shell32/wine/classes.c

```diff
@@ -334,6 +337,60 @@ BOOL HCR_GetIconA(LPCSTR szClass, LPSTR szDest, LPCSTR szName, DWORD len, int* p
 	return ret;
 }
 
+#ifdef __REACTOS__
+BOOL HCU_GetIconW(LPCWSTR szClass, LPWSTR szDest, LPCWSTR szName, DWORD len, int* picon_idx)
+{
+    HKEY hkey;
+    WCHAR sTemp[MAX_PATH];
+    BOOL ret = FALSE;
+
+    TRACE("%s\n", debugstr_w(szClass));
+
+    StringCchPrintfW(sTemp, _countof(sTemp), L"%s\\DefaultIcon", szClass);
+
+    if (!RegOpenKeyExW(HKEY_CURRENT_USER, sTemp, 0, KEY_READ, &hkey))
+    {
+        ret = HCR_RegGetIconW(hkey, szDest, szName, len, picon_idx);
+        RegCloseKey(hkey);
+	}
+
+    if (ret)
+        TRACE("-- %s %i\n", debugstr_w(szDest), *picon_idx);
+    else
+        TRACE("-- not found\n");
+
+    return ret;
+}
+
+BOOL HLM_GetIconW(int reg_idx, LPWSTR szDest, DWORD len, int* picon_idx)
+{
+    HKEY hkey;
+    WCHAR sTemp[5];
+    BOOL ret = FALSE;
+
+    TRACE("%d\n", reg_idx);
+
+    StringCchPrintfW(sTemp, _countof(sTemp), L"%d", reg_idx);
```

Comment:
Each digit in the number is converted to a character, so yes, if the number is longer than 5 digits, then it will create the string longer than 5 characters appropriately. **BUT** such long registry indexes simply don't exist, so they will never be passed to the function. The maximal length with 5 characters is used basing on maximum possible number of digits in registry index.